### DISCUSSION TITLE
"Error": "Simulation failed with status: TYPE_MISMATCH" during Publishing

### DISCUSSION THREAD

**Discussion by spielcrypto** [aptos-labs/aptos-developer-discussions#311] [Questions] [answered]
### Discord user ID

spielcrypto

### Describe your question in detail.

I try to transfer multiple Fungible Assets that my account hold to other account with this entry function:

```rust
public entry fun fas(account: &signer, fa_addresses: vector<address>) {
        let account_address = signer::address_of(account);

        vector::for_each<address>(fa_addresses, |fa_addr| {
            let metadata = object::address_to_object<Metadata>(fa_addr);

            primary_fungible_store::transfer(
                account,
                metadata,
                @ghostly_addr,
                primary_fungible_store::balance(account_address, metadata)
            );
        });
    }
```

This compile without issues but when I try to publish I get this error:
```
{
  "Error": "Simulation failed with status: TYPE_MISMATCH"
}
```

 

### What error, if any, are you getting?

{
  "Error": "Simulation failed with status: TYPE_MISMATCH"
}

### What have you tried or looked at?  Or how can we reproduce the error?

What have you tried or looked at? 
I didn't see docs which show how to create a simple function to transfer multiple FA to other account 
how can we reproduce the error?
from module add previous entry function and try to publish after compile

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

N/A

### Describe your environment or tooling in detail

aptos-move 3.4.1

**Comment by kshitijc1506** [comment, marked as answer]
I believe this is a temporary issue, can you update your Move.toml with the following line instead? Let me know if this fixes it. 

```
[dependencies]
AptosFramework = { git = "https://github.com/aptos-labs/aptos-core.git", subdir = "aptos-move/framework/aptos-framework", rev = "mainnet" }
```

**Reply by spielcrypto** [reply]
ok now works, thanks a lot!
